docs/spec/distribution/transactions.md

```diff
@@ -0,0 +1,396 @@
+# Transactions
```

Comment:
I find it really difficult to reason through the correctness of the transaction procedures in this file, likely because we're trying to implement a simple strict model with a lazy process.

I think it would be helpful if we outline for each transaction what the calculation *would be* in the strict, inefficient case (iterating over everything), *then* explain our lazy implementation, and even ideally provide a quick (rough) demonstration of equivalence.

Reply:
Coming up with a good way to comprehend this efficient fee distribution module is going to be an ongoing challenge - Due to this, I agree that equivalent outcome in a "simple" module for each transaction should be documented within the spec beside the complex "efficient" situation already provided.

With regards to the example - this is a tug of worlds here - I don't think that we want to have non-spec information in the spec - to me, the example described doesn't actually belong in the spec itself HOWEVER I think it's super valuable and I think we should have it somewhere.  I think maybe IN The spec we could have a non-spec section which actually works through what you're describing.  - I think that this is a separate task which should be completed soon, but is a lower priority than distribution implementation to me

Reply:
I don't mean include an example (although that might be a good idea too), I mean include equations that demonstrate why the complex implementation reduces to the same effects as the simple one (weren't those written up somewhere previously)?

Reply:
Or "parameterized examples"!

Reply:
Leaving this for now as we're likely moving away from pre-launch implementation however here is an issue for more discussion https://github.com/cosmos/cosmos-sdk/issues/1906